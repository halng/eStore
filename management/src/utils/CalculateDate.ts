export const calculateTimeAgo = (dateString: string): string => {
    const date = new Date(dateString)
    const currentDate = new Date()
    const timeDifference = currentDate.getTime() - date.getTime()

    const seconds = Math.floor(timeDifference / 1000)
    const minutes = Math.floor(seconds / 60)
    const hours = Math.floor(minutes / 60)
    const days = Math.floor(hours / 24)

    if (minutes < 1) {
        return `${seconds} seconds ago`
    } else if (hours < 1) {
        return `${minutes} minutes ago`
    } else if (days < 1) {
        return `${hours} hours ago`
    } else {
        return `${days} days ago`
    }
}
