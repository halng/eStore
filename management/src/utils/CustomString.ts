export const FormatString = (str: string, ...val: string[] | any[]) => {
    for (let index = 0; index < val.length; index++) {
        str = str.replace(`{${index}}`, val[index])
    }
    return str
}
