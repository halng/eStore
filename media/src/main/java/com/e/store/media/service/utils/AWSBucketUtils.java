package com.e.store.media.service.utils;

import com.amazonaws.auth.AWSCredentials;
import com.amazonaws.auth.AWSStaticCredentialsProvider;
import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;
import com.amazonaws.services.s3.model.CannedAccessControlList;
import com.amazonaws.services.s3.transfer.TransferManager;
import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
import com.amazonaws.services.s3.transfer.Upload;
import com.e.store.media.exception.InternalServiceException;
import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.springframework.web.multipart.MultipartFile;

@Component
public class AWSBucketUtils extends UploadToBucketUtils {

	private static final Logger LOGGER = LoggerFactory.getLogger(GCPBucketUtils.class);

	@Value("${aws.access.secret}")
	private String awsSecret;

	@Value("${aws.access.key}")
	private String awsKey;

	@Value("${aws.bucket.name}")
	private String awsBucket;

	@Value("${aws.bucket.region}")
	private String awsRegion;

	@Value("${aws.object.endpoint}")
	private String endPointUrl;

	private AmazonS3 amazonS3;

	public AWSBucketUtils() {
	}

	AmazonS3 getAmazonS3CLI() {
		AWSCredentials credentials = new BasicAWSCredentials(awsKey, awsSecret);
		return AmazonS3ClientBuilder.standard()
			.withCredentials(new AWSStaticCredentialsProvider(credentials))
			.withRegion(awsRegion)
			.build();
	}

	private File convertToFile(String fileName, MultipartFile multipartFile) throws IOException {
		File file = File.createTempFile(fileName, getFileExtension(multipartFile.getOriginalFilename()));

		try (FileOutputStream fos = new FileOutputStream(file)) {
			fos.write(multipartFile.getBytes());
		}
		catch (Exception e) {
			LOGGER.error("Error when converting Multipart to file with error msg: {}", e.getMessage());
		}

		return file;
	}

	private void putObjectToS3(TransferManager transferManager, String fileName, File file) {

		try {
			long start = System.currentTimeMillis();
			Upload result = transferManager.upload(awsBucket, fileName, file);
			result.waitForCompletion();
			long end = System.currentTimeMillis();
			LOGGER.info("Complete Multipart Uploading {}s", (end - start) / 1000);
		}
		catch (InterruptedException e) {
			LOGGER.error("Encountering error: {}", e.getMessage());
			// follow
			// https://stackoverflow.com/questions/54434976/either-re-interrupt-this-method-or-rethrow-the-interruptedexception-issue-in-so
			Thread.currentThread().interrupt();
		}
	}

	@Override
	public String uploadFile(MultipartFile multipartFile, String fileName, String contentType) {
		String fileUrl = "";
		try {
			String finalFileName = fileName + getFileExtension(multipartFile.getOriginalFilename());
			fileUrl = endPointUrl + "/" + awsBucket + "/" + finalFileName;
			File file = convertToFile(fileName, multipartFile);
			amazonS3 = getAmazonS3CLI();
			TransferManager transferManager = TransferManagerBuilder.standard()
				.withS3Client(amazonS3)
				.withMultipartUploadThreshold((long) (50 * 1024 * 1025))
				.build();

			putObjectToS3(transferManager, finalFileName, file);
			amazonS3.setObjectAcl(awsBucket, finalFileName, CannedAccessControlList.PublicRead);
		}
		catch (IOException e) {
			throw new InternalServiceException("Can not upload media file with error: " + e.getMessage());
		}
		return fileUrl;
	}

}
