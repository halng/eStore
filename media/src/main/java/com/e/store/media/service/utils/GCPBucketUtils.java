package com.e.store.media.service.utils;

import com.google.cloud.storage.Blob;
import com.google.cloud.storage.BlobId;
import com.google.cloud.storage.BlobInfo;
import com.google.cloud.storage.Storage;
import com.google.cloud.storage.StorageOptions;
import java.io.IOException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.springframework.web.multipart.MultipartFile;

//@Component
public class GCPBucketUtils extends UploadToBucketUtils {

    private static final Logger LOGGER = LoggerFactory.getLogger(GCPBucketUtils.class);


    @Value("${ggcloud.bucket.name}")
    private String BUCKET_NAME;

    @Value("${ggcloud.project-id}")
    private String PROJECT_ID;

    @Override
    public String uploadFile(MultipartFile multipartFile, String fileName, String contentType) {
        try {
            String fileExe = getFileExtension(multipartFile.getOriginalFilename());
            String finalFileName = fileName + fileExe;

            Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build()
                .getService();
            BlobId blobId = BlobId.of(BUCKET_NAME, finalFileName);
            BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType(contentType).build();
            Blob blob = storage.create(blobInfo, multipartFile.getBytes());
            if (blob != null) {
                LOGGER.info("File successfully uploaded to Storage");
                return new StringBuilder().append("https://storage.googleapis.com/")
                    .append(BUCKET_NAME).append("/").append(finalFileName).toString();
            }
        } catch (IOException e) {
            LOGGER.error("Can't get data from file " + fileName);
            return "";
        }
        return "";
    }

}
