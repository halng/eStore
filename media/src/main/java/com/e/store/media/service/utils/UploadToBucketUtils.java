package com.e.store.media.service.utils;

import com.e.store.media.exception.BadRequestException;
import java.util.Arrays;
import java.util.List;
import org.springframework.stereotype.Component;
import org.springframework.web.multipart.MultipartFile;

@Component
public abstract class UploadToBucketUtils {

	private final List<String> SUPPORTED_EXTENSION = Arrays.asList(".jpeg", ".jpg", ".png");

	String getFileExtension(String fileName) {
		if (fileName != null && fileName.contains(".")) {
			for (var e : SUPPORTED_EXTENSION) {
				if (fileName.endsWith(e)) {
					return e;
				}
			}
		}
		throw new BadRequestException("File extension not supported " + fileName);
	}

	public abstract String uploadFile(MultipartFile multipartFile, String fileName, String contentType);

}
