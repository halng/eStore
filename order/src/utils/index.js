module.exports.simpleTest = (a, b) => {
    return a * b
}
