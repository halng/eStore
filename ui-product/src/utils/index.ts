import slugify from 'slugify'

const convertStandardSlug = (value: string): string => {
    if (value) {
        return slugify(value, {
            remove: /[*+~.()'"!:@]/g,
            replacement: '-',
            lower: true,
            trim: true,
        })
    }
    return ''
}

export { convertStandardSlug }
