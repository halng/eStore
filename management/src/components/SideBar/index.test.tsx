import '@testing-library/jest-dom'
import { screen } from '@testing-library/react'
import { SideBar } from '@components'
import navConfig from './config-navigation'
import { renderWithProviders } from '../../utils/test-utils'
import { AuthState } from '@types'

jest.mock('next/navigation', () => ({
    usePathname: jest.fn(() => '/partner'),
}))

describe('Test SideBar Component', () => {
    it('renders a side bar', () => {
        const initData: AuthState = {
            isAuth: true,
            username: 'Tao Nguyen',
            photoUrl: 'https://www.google.com',
            role: 'SELLER',
            email: 'hello@gmail.com',
            id: '123',
        }
        const { container } = renderWithProviders(<SideBar />, {
            preloadedState: { auth: initData },
        })

        // check render account
        const accountName = screen.getByText('Tao Nguyen')
        const accountRole = screen.getByText('SELLER')
        expect(accountName).toBeInTheDocument()
        expect(accountRole).toBeInTheDocument()

        // check render menu
        const menuItems = container.querySelectorAll('a')
        expect(menuItems).toHaveLength(navConfig.length)
        for (let i = 0; i < navConfig.length; i++) {
            expect(menuItems[i]).toHaveTextContent(navConfig[i].title)
        }
    })
})
