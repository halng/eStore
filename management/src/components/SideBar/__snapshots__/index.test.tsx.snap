// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SideBar Component snapshot rendering 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1y6sui9"
  >
    <div
      class="MuiBox-root css-mf19sy"
    >
      <div
        class="MuiBox-root css-1pdmvny"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-17o22dy-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          class="MuiBox-root css-d0uhtl"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          />
        </div>
      </div>
      <nav
        class="MuiStack-root css-hv5cso-MuiStack-root"
      />
      <div
        class="MuiBox-root css-i9gxme"
      />
    </div>
  </div>
</DocumentFragment>
`;
